wordpress-mobile/WordPress-Aztec-iOS
# Adds node descriptors.

This is part of my cleanup work before I start implementing undo/redo.  The reason I'm tackling this is that I want to make sure our methods don't pass an enormous amount of parameters around.

This PR introduces the concept of node descriptors to simplify `ElementNode` searching and creation, while simplifying our calls.

**Known issues:**

While implementing this feature I noticed an issue with styles not being properly removed from the HTML source anymore.  This issue however was not introduced by this PR, so I opened a separate issue to fix it.

I uploaded a video of the problem [here](https://cloudup.com/cngEAn9uNba).

**How to test:**

Test the editor normally.  Try to come up with edge scenarios and make sure nothing breaks.
Run the unit tests.
